zstd/decoder.go

```diff
@@ -156,8 +156,14 @@ func (d *Decoder) Reset(r io.Reader) error {
 	if d.current.err == ErrDecoderClosed {
 		return d.current.err
 	}
+
+	d.drainOutput()
+
 	if r == nil {
-		return errors.New("nil Reader sent as input")
+		d.current.b = []byte{}
+		d.current.err = io.EOF
```

Comment:
This would make a read seem like the payload was a valid 0 byte block. 

We want to make it clear that a read was performed with a nil input.

```Go
	// ErrNilInput is returned when a nil Reader was given
	// and a non-Reset operation was attempted. 
	ErrNilInput = errors.New("nil input provided as reader")
```

```suggestion
		d.current.err = ErrNilInput
```

Reply:
I guess we should also return this sentinel error if the Decoder was created with a nil reader.

Reply:
Added this change, but with the name ErrDecoderNilInput in the fixup. Maybe that's too specific though?

Reply:
It's fine. Thanks!